Apps/Common/src/Delegates/WalletIssuerDelegate.cs

```diff
@@ -119,10 +123,81 @@ public async Task<RequestResult<ConnectionResponse>> CreateConnectionAsync(Guid
         }
 
         /// <inheritdoc/>
-        public Task<RequestResult<CredentialResponse>> CreateCredentialAsync<T>(WalletConnection connection, T payload)
-            where T : CredentialPayload
+        public async Task<RequestResult<CredentialResponse>> CreateCredentialAsync<T>(WalletConnection connection, CredentialPayload payload, string comment)
         {
-            throw new NotImplementedException();
+            _ = connection.AgentId ?? throw new ArgumentException("AgendId of Connection cannot be null");
```

Comment:
?

Reply:
We can chat about this during the PR discussion.

Reply:
it looks terribly odd. We should use the regular if and throw if necessary instead of discarding this